### ISSUE TITLE
FSxN - Error creating file security permissions acl - Code 4 - Entry doesn't exist

### ISSUE THREAD

**Issue by PavGregor** [ansible-collections/netapp.ontap#146]
### Summary

Hi, I'm using the latest `netapp.ontap` collection (22.4.1) to manage an FSxN filesystem.

I am seeing the following error when trying to modify the NTFS ACL permissions using the `netapp.ontap.na_ontap_file_security_permissions_acl` module:

```
{'message': \"entry doesn't exist\", 'code': '4', 'target': 'uuid'}
```

Note: checking the cifs share afterwards I can confirm the changes are actually applied to the share, the correct users/permissions are configured, as long as I set Ansible to ignore the error and carry on with the playbook execution.

### Component Name

netapp.ontap.na_ontap_file_security_permissions_acl

### Ansible Version

```console
$ ansible --version
ansible [core 2.12.5.post0]
  config file = None
  configured module search path = ['/home/runner/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/runner/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.13 (default, Jun 24 2022, 15:27:57) [GCC 8.5.0 20210514 (Red Hat 8.5.0-13)]
  jinja version = 2.11.3
  libyaml = True
```

### ONTAP Collection Version

```console
$ ansible-galaxy collection list
# /usr/share/ansible/collections/ansible_collections
Collection        Version
----------------- -------
amazon.aws        5.2.0
community.crypto  2.8.1
community.general 4.8.1
netapp.ontap      22.4.1
```

### ONTAP Version

```console
fsxn::> version
NetApp Release 9.11.1P6: Fri Jan 27 03:16:05 UTC 2023
```

### Playbook

```console
- name: Modify NTFS ACL on FSxN share
  hosts: localhost
  gather_facts: no

  tasks:
	- name: NTFS ACL
	  netapp.ontap.na_ontap_file_security_permissions_acl:
	    vserver: fsxn_fs_svm
	    hostname: fsxn_share_host
	    username: fsxadmin
	    password: '{{ fsxadmin_password }}'
	    validate_certs: false
	    use_rest: always
	    state: present
	    path: /test
	    validate_changes: warn
	    access_control: file_directory
	    access: access_allow
	    acl_user: adminuser
	    rights: full_control
	    apply_to:
	      this_folder: true
	      sub_folders: true
	      files: true
	  register: acl_task_output
```

### Steps to Reproduce

Use the above playbook to modify the NTFS ACL permissions on an existing cifs share.

### Expected Results

The task should succeed without errors.

### Actual Results

```console
{
  "exception": "NoneType: None\n",
  "msg": "Error creating file security permissions acl /test: job reported error: {'message': \"entry doesn't exist\", 'code': '4', 'target': 'uuid'} - {'message': \"entry doesn't exist\", 'code': '4', 'target': 'uuid'} - {'message': \"entry doesn't exist\", 'code': '4', 'target': 'uuid'} - {'message': \"entry doesn't exist\", 'code': '4', 'target': 'uuid'}, received {'job': {'uuid': '1df70c27-cd41-11ed-a36d-69c975918b02', '_links': {'self': {'href': '/api/cluster/jobs/1df70c27-cd41-11ed-a36d-69c975918b02'}}}}.",
  "invocation": {
    "module_args": {
      "vserver": "fsxn_fs_svm",
      "hostname": "fsxn_share_host",
      "username": "fsxadmin",
      "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
      "validate_certs": false,
      "use_rest": "always",
      "state": "present",
      "path": "/test",
      "validate_changes": "warn",
      "access_control": "file_directory",
      "access": "access_allow",
      "acl_user": "adminuser",
      "rights": "full_control",
      "apply_to": {
        "this_folder": true,
        "sub_folders": true,
        "files": true
      },
      "https": false,
      "http_port": null,
      "ontapi": null,
      "feature_flags": null,
      "cert_filepath": null,
      "key_filepath": null,
      "force_ontap_version": null,
      "advanced_rights": null,
      "ignore_paths": null,
      "propagation_mode": null
    }
  },
  "warnings": [
    "This module is not idempotent when \"rights\" is used, make sure to use \"advanced_rights\"."
  ],
  "_ansible_no_log": false,
  "changed": false,
  "failed_when_result": false,
  "item": "adminuser",
  "ansible_loop_var": "item",
  "_ansible_item_label": "adminuser",
  "_ansible_delegated_vars": {
    "ansible_host": "localhost",
    "ansible_port": null,
    "ansible_user": "ansibleuser",
    "ansible_connection": "local"
  }
}
```

**Comment by carchi8py** [comment]
This is a bug currently with FSX that we are looking for a workaround for

**Comment by carchi8py** [comment]
Just an update this bug appear to be on the ONTAP side, we are waiting on the ONTAP team to have a public bug report for this.

**Comment by PavGregor** [comment]
ok thank you for the update, any ETA when a fix or workaround might be available?

**Comment by carchi8py** [comment]
I don't know if the Burt is public yet (Not part of the ONTAP Team). But let me give you a short update on what the issue is.

For any job that is Async in ONTAP a Job is returned.  
With FSx only (doesn't effect non FSx system) the fsxadmin does not have permission to view the job.  The Job contain when the task is done, if there is an error, and thing of that nature.  Without that information there no way for Ansible to move forward. 

So what going on above is we are calling that Job 4 times and each time we get a 
{'message': \"entry doesn't exist\", 'code': '4', 'target': 'uuid'}

After the 4 attempts, we give up and return the error.

The ONTAP team has a burt open to fix this. (again since i don't know if it public yet i can't share the burt info yet)

**Comment by PavGregor** [comment]
@carchi8py great thanks for the update. 

Unfortunately my FSxN file systems in AWS are currently on `NetApp Release 9.13.0P4` so I'll need to wait until the next update cycle to be able to test the fix.

**Comment by PavGregor** [comment]
@carchi8py I not sure this is actually fixed for FSxN, at least not with the latest released version:
```
::> version
NetApp Release 9.13.1P5: Thu Nov 02 20:37:09 UTC 2023 
```
```
# ansible-galaxy collection list
Collection        Version
----------------- -------
community.general 7.2.1  
netapp.ontap      22.9.0
```
If it's of any help, it seems to work with `acl_user: Everyone`, although the API takes a while (roughly 20-25s) to return.
However with `acl_user: some_AD_user`, the API executes without issues only if the user doesn't have an ACL entry yet on the target share, but if it does I get this error:
```
The full traceback is:
NoneType: None
[WARNING]: This module is not idempotent when "rights" is used, make sure to use "advanced_rights".
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "access": "access_allow",
            "access_control": "file_directory",
            "acl_user": "some_AD_user",
            "advanced_rights": null,
            "apply_to": {
                "files": true,
                "sub_folders": true,
                "this_folder": true
            },
            "cert_filepath": null,
            "feature_flags": {
                "trace_apis": true,
                "trace_headers": true
            },
            "force_ontap_version": null,
            "hostname": "fsx_hostname",
            "http_port": null,
            "https": false,
            "ignore_paths": null,
            "key_filepath": null,
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "path": "/ansible_share",
            "propagation_mode": null,
            "rights": "full_control",
            "state": "present",
            "use_rest": "always",
            "username": "fsxadmin",
            "validate_certs": false,
            "validate_changes": "warn",
            "vserver": "fsx_vserver"
        }
    },
    "msg": "Error creating file security permissions acl /ansible_share: calling: protocols/file-security/permissions/<id>/%2Fansible_share/acl: got {'message': 'duplicate entry', 'code': '1', 'target': 'user'}."
}
```

**Comment by carchi8py** [comment]
Let me check what going on.

This is a different issue from the previous one.  The original issues was that we couldn't get access to the Job 

{'message': \"entry doesn't exist\", 'code': '4', 'target': 'uuid'}) 

Let me check what going on. (I think I vaguely remember seeing a similar issue on the non-FSX side of ONTAP, where sometimes an ACL is written as a single entry. And sometimes, it is broken up into 2 entries.

**Comment by PavGregor** [comment]
@carchi8py after some more testing I'm pretty sure that the "code 4" issue is indeed resolved.

The issue I hit upon when testing this was that I was using `acl_user: AD_user` instead of  `ACL_user: DOMAIN\AD_user`. The confusing thing is that leaving out the domain the `na_ontap_file_security_permissions_acl` task is still able to create a new ACL entry, but it errors out if you try to modify or remove it.

**Comment by suhasbshekar** [comment]
If this is resolved, can we close this issue?
